src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,394 @@
+     * <li>{@link org.bouncycastle.pkcs.PKCS10CertificationRequest}
+     * <li>{@link org.bouncycastle.jce.provider.X509CertificateObject}
+     * <li>{@link org.bouncycastle.x509.X509V2AttributeCertificate}
+     * </ul>
+     * </ul>
+     * 
+     * @return {@link Object} read from the PEM
+     */
+    @Nullable
+    public Object getRawObject() {
+        return pemObject;
+    }
+
+    /**
+     * Obtains the fingerprint of the private key in the "ab:cd:ef:...:12" format, which basically is an SHA1 digest
+     * from the key, encoded in hex format.
+     * 
+     * @return private key fingerprint in hex format "ab:cd:ef:...:12", null if the private key could not be obtained
+     * from the current PEM data.
+     * @throws IOException thrown if a problem exists creating the fingerprint
+     */
+    @Nullable
+    public String getPrivateKeyFingerprint() throws IOException {
+        PrivateKey key = getPrivateKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestSHA1(key));
+    }
+
+    /**
+     * Obtains the fingerprint of the public key in the "ab:cd:ef:...:12" format, which basically is an MD5 digest from
+     * the key, encoded in hex format.
+     * 
+     * @return public key fingerprint in hex format "ab:cd:ef:...:12", null if the public key could not be obtained from
+     * the current PEM data.
+     * @throws IOException if a problem exists creating the fingerprint
+     */
+    @Nullable
+    public String getPublicKeyFingerprint() throws IOException {
+        PublicKey key = getPublicKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestMD5(key));
+    }
+
+    /**
+     * Generates an SHA1 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     */
+    @Nonnull
+    public static byte[] getKeyDigestSHA1(@Nonnull Key k) throws IOException {
+        return getKeyDigest(k, "SHA1");
+    }
+
+    /**
+     * Generates an MD5 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     */
+    @Nonnull
+    public static byte[] getKeyDigestMD5(@Nonnull Key k) throws IOException {
+        return getKeyDigest(k, "MD5");
+    }
+
+    /**
+     * Generates an digest from a Key object in the specified digest format. The supported digest formats will depend on
+     * the JVM API.
+     * 
+     * @param k key to generate the digest from
+     * @param dg digest format
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     */
+    @Nonnull
+    public static byte[] getKeyDigest(@Nonnull Key k, @Nonnull String dg) throws IOException {
```

Comment:
this one should throw the `NoSuchAlgorithmException` as the algorithm is user supplied (and call `dg` as `algorithm`)